### ISSUE TITLE
Use a Convention insteat of Reflection

### ISSUE THREAD

**Issue by richardliebmann** [richardlawley/EntityFrameworkAttributeConfig#3]
Hi Richard,

first off all: This is a very great Library!

After some troubles with inheritance and generic types, i have found a nice way to implement this:

```
using System.Data.Entity.ModelConfiguration.Configuration;
using System.Data.Entity.ModelConfiguration.Conventions;

public class DecimalPrecisionAttributeConvention : PrimitivePropertyAttributeConfigurationConvention<DecimalPrecisionAttribute>
{
    public override void Apply(ConventionPrimitivePropertyConfiguration configuration, DecimalPrecisionAttribute attribute)
    {
        configuration.HasPrecision(attribute.Precision, attribute.Scale);
    }
}
```

and then in the OnModelCreation:

```
modelBuilder.Conventions.Add(new DecimalPrecisionAttributeConvention());
```

What do you think about this solution?

**Comment by richardlawley** [comment]
Richard,

This looks like a much neater solution, and avoids the need for Reflection.  Just working out how best to add this breaking change in, but I plan on updating the library to use this.

Thanks!
Richard

**Comment by richardlawley** [comment]
I've released this as v1.0 and updated nuget.
